app/models/client.rb

```diff
@@ -382,8 +386,20 @@ def self.possible_availability_states
     return states
   end
 
+  def parked?
+    prevent_matching_until.present? && prevent_matching_until > Date.today
+  end
+
+  def available?
```

Comment:
This will want to have a different name, maybe `available_for_matching?`.  There's a column on client `available` which magically responds to `available?`